esmf_regrid/experimental/unstructured_scheme.py

```diff
@@ -26,9 +35,26 @@ def _cube_to_MeshInfo(cube):
 
 
 def _cube_to_GridInfo(cube):
+    # This is a simplified version of an equivalent function/method in PR #26.
+    # It is anticipated that this function will be replaced by the one in PR #26.
+    #
     # Returns a GridInfo object describing the horizontal grid of the cube.
     # This may be inherited from code written for the rectilinear regridding scheme.
-    pass
+    lon = cube.coord("longitude")
+    lat = cube.coord("latitude")
+    # Ensure coords come from a proper grid.
+    assert isinstance(lon, iris.coords.DimCoord)
+    assert isinstance(lat, iris.coords.DimCoord)
```

Comment:
I like this - but I've just had a thought as to whether the scalar passes.  Let's just confirm, and remove these lines if they are more problematic than helpful!

Reply:
The scalar test is passing fine with this.
Having discussed this with @stephenworsley , we think this is because the information is added to the `Cube` as a DimCoord, then it becomes a scalar as it has length 1. Hence the test and the data process correctly.